tests/gtests/test_eltwise.cpp

```diff
@@ -164,20 +170,62 @@ void check_eltwise_fwd(const eltwise_test_params &p,
     }
 }
 
+template <typename data_t> data_t fwd_eps(algorithm alg)
+{
+    UNUSED(alg);
+    return 0;
+}
+
+template<> float fwd_eps<float>(algorithm alg)
+{
+    return (alg == algorithm::eltwise_elu) ? 2e-5f
+        : (alg == algorithm::eltwise_soft_relu) ? 2e-6f
+        : 1e-6f;
+}
+
+static const float FWD_EPS_F16 = 5e-2f;
+
+template<> bfloat16_t fwd_eps<bfloat16_t>(algorithm alg)
+{
+    UNUSED(alg);
+    return FWD_EPS_F16;
+}
+
+template<> float16_t fwd_eps<float16_t>(algorithm alg)
+{
+    UNUSED(alg);
+    return FWD_EPS_F16;
+}
+
 template <typename data_t>
 void compare_eltwise_fwd(const eltwise_test_params &p,
         const memory::desc &md, const memory &dst, const memory &ref_dst)
 {
-    const data_t eps = (data_traits<data_t>::data_type == memory::data_type::f16
-            || data_traits<data_t>::data_type == memory::data_type::bf16)
-            ? 5e-2
-            : (p.alg_kind == algorithm::eltwise_elu)
-                ? 2e-5
-                : (p.alg_kind == algorithm::eltwise_soft_relu) ? 2e-6 : 1e-6;
+    compare_data(ref_dst, dst, fwd_eps<data_t>(p.alg_kind));
+}
+
+template <typename data_t> data_t bwd_eps(algorithm alg)
```

Comment:
This template function, seems  overkill, since `bwd_eps` is only dependent on `alg`